### PR TITLE
feat: add port ids

### PR DESCRIPTION
**Pull request by imrishabh18**
Closes - https://github.com/tscircuit/tscircuit/issues/291

Screenshot 2024-07-29 at 7 04 28 PM

### CODE REVIEW

File: tests/trace-builder/simple-trace-test.test.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    )
+    .add("trace", (tb) =>
+      tb.setProps({
+        from: ".R1 > .right",
+        to: ".R2 > .left",
+      })
+    )
+    .build()
+
+  await logSoup(soup)
```
Comment (seveibar) [inline tests/trace-builder/simple-trace-test.test.ts:52]: Can you add a check here, eg su(soup).pcb_trace.list()[0].trace.some(a => a.start_pcb_port_id)

**Review by seveibar — APPROVED** [review]

**Review by seveibar — CHANGES_REQUESTED** [review]
Oh looking at the screenshot: only the first and last point in the route should have the start_pcb_port_id, basically if the x/y is on the pcb port then the id should be defined.

**Comment by imrishabh18** [comment]
So, only the first and last endpoint will be having `start_pcb_port_id` and no `end_pcb_port_id`, while the other will be having both, right?

**Comment by seveibar** [comment]
No the others will have undefined. It's kind of confusing and likely to change but you should only have an idea if the x y position represents the port

**Comment by imrishabh18** [comment]
This is what you mean't, that the first one is having it's `port_id` as the starting point and similarly the last one is having it's `port_id` as the starting point (bi-directional)

**Review by seveibar — APPROVED** [review]

**Pull request merged**